### PR TITLE
docs: explain where pageProps come from in Provider docs

### LINKED ISSUE #1132: Explain how to populate Provider in _app
In the current version of the docs https://next-auth.js.org/getting-started/client#provider we do not explain how the `pageProps.session` prop is populated. It is not something that comes magically by `next-auth` as explained here: https://github.com/nextauthjs/next-auth/issues/1130#issuecomment-761689414

Simply adding a `Provider` in the `_app` with a `session={pageProps.session}` file won't magically add any session. The props provided to the `_app` component must come from somewhere. In this case, either from `getInitialProps` or `getServerSideProps`. Do note that this does not have to be in the `_app` file, so you don't necessarily opt-out of any Next.js optimization. Add it only to those pages that need auth.

Eg.:

```js
// pages/index.js
import {getSession} from "next-auth/client"
...
export async function getServerSideProps(ctx) {
  return {
    props: {
      session: await getSession(ctx)
    }
  }
}
...
```

This then will be available through `pageProps.session`, and this is the value you have to provide to the `Provider`. Thanks to this, any subsequent `useSession()` hook calls will only get the client-cached session value, instead of re-visiting the backend. You can configure how long you want to cache the session before getting the new one from the backend. Documented here: https://next-auth.js.org/getting-started/client#options

It would be nice to incorporate this into the docs, to avoid further confusion.

### PR DESCRIPTION
**Pull request by ndom91**
## Reasoning 💡

Further explain how the `pageProps` get into the page component and how the session prop actually gets passed through, i.e. through the use of Next.js `getInitialProps` and our `getSession` function, for example. 

## Checklist 🧢

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged
      

## Affected issues 🎟

Closes #1132

### CODE REVIEW

File: www/docs/getting-started/client.md [hunk 1]
```diff
@@ -328,6 +328,24 @@ export default function App ({ Component, pageProps }) {
...
+export async function getServerSideProps(ctx) {
+  return {
+    props: {
+      session: await getSession(ctx)
+    }
+  }
+}
+```
+
+If every one of your pages needs to be protected, you can do this in `_app`, otherwise you can do it on a page-by-page basis.
```
Comment (balazsorban44) [inline www/docs/getting-started/client.md:347]: Don't know if it's a true alternative, but the method described here https://github.com/nextauthjs/next-auth/issues/1210#issuecomment-782630909 makes it possible to do per page auth much easier without gSSP, and you can even show a fallback ui or extend it with roles or anything really.

gSSP in _app always felt very harsh to me, but I can see it's easier. What do you think?

Comment (ndom91) [reply www/docs/getting-started/client.md:347]: So the method you've linked looks really easy to implement with some nice conveniences tacked on, but I have to admit im not familiar enough with our internal session handling currently to fully appreciate the difference haha

I can definitely mention it in the docs though :+1:

Comment (balazsorban44) [reply www/docs/getting-started/client.md:347]: So it boils down to that if you use gSSP on the _app component, it means EVERY page transition will ALWAYS be blocking (meaning SSR), versus my suggestion, first you show a skeleton while loading/checking auth, and every page transition will be client-side afterwards.

**Comment by ndom91** [comment]
@balazsorban44 I've added an "Alternatives" section to the `getting-started/client.md` and referenced (1) your aforementioned custom client side auth check and (2) your `nextauthjs/react-query` library. 

I hope I got all the details right :sweat_smile: haha

**Pull request merged**